app/command_utils.cc.template

```diff
@@ -65,7 +68,23 @@ std::string GetCommandName(int command_id) {
          "commands (i.e. commands in |GetCommands()|). "
          "This command should probably be added.";
 
-  return l10n_util::GetStringUTF8(it->second);
+  // Matches &<key> and (&<key>)
+  static const base::NoDestructor<re2::RE2> kAmpersandPattern("(\\(?&[^\\s]\\)?)");
+
+  // Remove and '&' characters which don't have a space before and after.
+  auto translation = l10n_util::GetStringUTF8(it->second);
+
+  std::string match;
+  if (re2::RE2::PartialMatch(translation, *kAmpersandPattern.get(), &match)) {
+    // There are two cases here:
+    // 1. The & refers to a character in the string - in this case, we just
+    //    remove the '&'. For example "Install &PWA".
+    // 2. The & is in brackets, and defines a shortcut key not in the string, in
+    //    which case we remove the whole match. For example "Install (&K)PWA".
+    auto remove = base::StartsWith(match, "(") ? match : "&";
+    base::RemoveChars(translation, remove, &translation);
```

Comment:
Wouldn't this remove all &s in something like `Download & &Install`?

Reply:
I haven't tested, but maybe if we do a FullMatch with a pattern like `([^\(]*)(\(?&([\S])\)?)(.*)` and then grab group1 + group2.StartsWith("(") ? "" : group3 + group4?

Reply:
I was a little worried about this at first too, but it fits in with what Chromium is doing with App Menu items. Basically, these `&` characters are only used to indicate a shortcut key in App Menus, and the app menu strips out all `&` characters:

![image](https://github.com/brave/brave-core/assets/7678024/7ed4bd4a-ece4-4855-ae09-ad26c072bf7c)

from this string:

```
&amp;Downloads &amp; Settings
```

So I think this is fine as is because:
1. App Menu translations can't have `&` except to indicate a keyboard code (all &amp;'s are removed).
2. Other strings won't have an `&` next to a non-whitespace character.

Reply:
Got it. In that case this should work well. Thanks!